### PR TITLE
[1LP][RFR]Adding new file which includes test case related to quota

### PR DESCRIPTION
**Pull request by anurag03**
Purpose or Intent
=================
Implementing test case related to quota using automate method.

{{pytest: cfme/tests/infrastructure/test_quota.py -v}}

### CODE REVIEW

**Comment by dajoRH** [comment]
Lint report for commit ed9843a8278fd404d946480c8220ed82bb862fb5:

`cfme/tests/infrastructure/test_quota.py`:
- :bangbang: Line 115:15: **E702** *multiple statements on one line (semicolon)*

Please, rectify these issues :smirk: .


**No commit flaws detected**.

*CFME QE Bot*

**Comment by anurag03** [comment]
PRT failed because it didn't get the 5.8 appliance, checked locally, it's working fine hence changed the state from WIPTEST to RFR.

File: cfme/tests/infrastructure/test_quota.py [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+        classes.instantiate('QuotaStateMachine'). \
+        instances.instantiate('quota')
+    original_instance.copy_to(domain=test_domain)
+
+    instance = test_domain. \
+        namespaces.instantiate('System'). \
+        namespaces.instantiate('CommonMethods'). \
+        classes.instantiate('QuotaStateMachine'). \
+        instances.instantiate('quota')
+    yield instance
```
Comment (rlbabyuk) [inline cfme/tests/infrastructure/test_quota.py]: Required: use return here, as we don't have any cleanup or further actions with instance

File: cfme/tests/infrastructure/test_quota.py [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+    with update(max_quota_test_instance):
+        max_quota_test_instance.fields = {'quota_source_type': {'value': entity}}
+
+
+@pytest.fixture(params=[('groups', 'group', 'EvmGroup-super_administrator'),
+                        ('users', 'user', 'Administrator')], ids=['group', 'user'], scope='module')
+def entities(appliance, request, max_quota_test_instance):
+    collection, entity, description = request.param
+    set_entity_quota_source(max_quota_test_instance, entity)
+    yield getattr(appliance.collections, collection).instantiate(description)
```
Comment (rlbabyuk) [inline cfme/tests/infrastructure/test_quota.py]: Required: use return here

File: cfme/tests/infrastructure/test_quota.py [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+
+@pytest.fixture
+def template_name(provider):
+    if provider.one_of(RHEVMProvider):
+        return provider.data.templates.get('full_template')['name']
+    elif provider.one_of(VMwareProvider):
+        return provider.data.templates.get('big_template')['name']
+
+
+@pytest.fixture()
```
Comment (rlbabyuk) [inline cfme/tests/infrastructure/test_quota.py]: Required: () is  not necessary here

File: cfme/tests/infrastructure/test_quota.py [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+    ['custom_prov_data'],
+    [
+        [{'hardware': {'memory': '4096'}}],
+        [{}],
+        [{'hardware': {'vm_num': '21'}}],
+        [{'hardware': {'num_sockets': '8'}}]
+    ],
+    ids=['max_memory', 'max_storage', 'max_vm', 'max_cpu']
+)
+def test_quota(appliance, provider, setup_provider, custom_prov_data, vm_name, admin_email,
```
Comment (rlbabyuk) [inline cfme/tests/infrastructure/test_quota.py]: Required: Add test description

**Comment by anurag03** [comment]
Sprout did not provision appliance for 5.8 and hence the test failed for 5.8 hence working fine locally. thus changing the state from WIPTEST to RFR.
@rlbabyuk you can review it now.

**Pull request merged**